### ISSUE TITLE
Documentation / Wiki

### ISSUE THREAD

**Issue by bell07** [Gerold55/laptop#48]
I think we need documentation for dev's and for users.
The dev's docu is the API.md in repository. For users I propose to write small documentation on for hardware, system and existing apps from users perspective

**Comment by Gerold55** [comment]
Agreed. How should it be setup though? Should it read like an actual laptop manual or just as a mod description?

**Comment by bell07** [comment]
Laptop manual could be an ingame app oO ;-)
I think it should be an overview as showcase for mod capabilities

**Comment by Gerold55** [comment]
:)

**Comment by Crosszover** [comment]
![laptop_manual](https://user-images.githubusercontent.com/33940608/33914261-c09382ec-df73-11e7-8cc0-8e3cc714c8ee.png)
 ingame user manual available :)

**Comment by bell07** [comment]
;-) Thanks.. But only the icon, the content still missing ;-(

**Comment by bell07** [comment]
What you are thinking about the "Help" app that just display the content of existing "README.txt" ? So the player can see the new version ingame and some other basic informations.

**Comment by Gerold55** [comment]
Under the documentation/wiki needs to be a crafting guide for this mod and add-ons. I hope we can get this part of the mod working soon so we'll be able to make the recipes soon.

**Comment by Gerold55** [comment]
@Grizzly-Adam what do you think?

**Comment by bell07** [comment]
The initial idea for this usse was more documentation about the mod for users or developers using github wiki and/or pages.

You mean ingame help like the Wuzzy's documentation system?
 https://forum.minetest.net/viewtopic.php?f=9&t=15912

and/or https://forum.minetest.net/viewtopic.php?f=11&t=14088

Such mods are usually active in players inventory, if installed so there is no added value if we have them as laptop app

**Comment by Gerold55** [comment]
I still like the idea of the recipes and about the computer being in the game as a book/wiki

**Comment by Grizzly-Adam** [comment]
Other than crafting recipes (which we don't have) what information are you thinking of adding? It all seems pretty self explanatory other than CS-BOS which has its own help system built in.

**Comment by bell07** [comment]
Agree with crafting guide, the information could be generated.
 But additional documentation needs to be written.
 Do you like to see a generic guide about all items found in system or a guide that shows laptop related data only? Maybe some "factoids" for doc-system?

**Comment by Grizzly-Adam** [comment]
Ah, I could write a quirky manual for the end users.

**Comment by Gerold55** [comment]
The recipes are currently being worked on by @Crosszover & I we have some images of components for the crafting recipes.

**Comment by Grizzly-Adam** [comment]
I have begun writing a manual for CS-BOS. Also, I writing it in CS-BOS! I will also cover the "shell" in that file.

Do we have a name for our GUI? That will be my next target for documentation.

**Comment by bell07** [comment]
Do we need an help app that reads appname_help.txt-file from mod directory and show them in laptop? And a help button in window bar for each app?

**Comment by bell07** [comment]
I called the ui "ModernOS" or "LaptopOS" but there is no official names. So you can take proposals too

**Comment by Grizzly-Adam** [comment]
I think that an in-game help button would be good, and one for the GUI as well. I mocked up a couple pictures of where I think the help button should be placed. In the GUI, I think it should become either part of the clock element or be able to be placed anywhere the theme needs it to be, like we did with the clock. I think the later would probably be better. 

![app-help](https://user-images.githubusercontent.com/35204361/37255200-06d19418-2517-11e8-9272-87de6d4dfca9.png)
![gui-help](https://user-images.githubusercontent.com/35204361/37255201-06f3ddf2-2517-11e8-8d7e-3e6948013192.png)

**Comment by bell07** [comment]
Agree the help button on launcher should be themes-configurable like the clock.

The position like you propose for text editor cannot proper applied to all apps. My idea was to add the "?" button to the window bar like "back" and "exit" buttons, and manage the functionality fully in launcher. If a "appname_help.txt" exists, show the button...

**Comment by Grizzly-Adam** [comment]
The CS-BOS manual is created and released as a PDF. Do we want to make it an in-game book as well?

**Comment by Grizzly-Adam** [comment]
re: "?" on menu bar. Pretty soon we will end up needing an actual menu. Maybe we should stop a moment and think if we will be needing to add anything us in the near future.

**Comment by bell07** [comment]
Drop-down menus are to dynamic for formspecs :-( There is a simple "drop down" but looks not nice..

Do we really need Help and about? We cannot take "popup window" for about. What should "RUN" do?
Trying to keep it simple, may I should just add a mtos:show_help_dialog('app_name') and the help button needs to be added to each app that supports help on individual place.

Where I can see the PDF?

**Comment by Grizzly-Adam** [comment]
We really need help. The others was me just trying to be forward thinking. If its just the help, we cwn do the ? Wediscussed before. 

https://github.com/Grizzly-Adam/Minetest-Laptop-Manuals

**Comment by bell07** [comment]
I meant do we need both? Agree the "help" is usefull, but unsure about "About".
The manual looks good ;-) 
Can you add the "pdf-source" to the github too?
Unsure if `DIR FDD` does work in the last version. I think I changed it to "VFS-Style", that means `DIR FDD:` or `DIR FDD:*` (or dos-like `DIR FDD:*.*`). The prevous DIR FDD search for a file called FDD :-(

**Comment by Grizzly-Adam** [comment]
Skip about, just do help. I will upload the odt in the near future.

**Comment by Gerold55** [comment]
Should we have a wiki/documentation app?

**Comment by Gerold55** [comment]
I think that there should be the question mark but how would we get it to go to the correct page of the wiki?

**Comment by Grizzly-Adam** [comment]
I am thinking we just hard code each app to look up its own info.

**Comment by Gerold55** [comment]
Yeah we could do that also.

**Comment by bell07** [comment]
> I think that there should be the question mark but how would we get it to go to the correct page of the wiki?

It is possible to write "selected_page" to the app data before calling the app

**Comment by bell07** [comment]
Did some draft work for testing and tech-demo in my [docu_app branch](https://github.com/bell07/minetest-laptop/tree/docu_app).
The dialog is an app, called from launcher the app show text from apps/launcher_doc.txt file.
From other apps the call is possible as implemented in file selection dialog:
```
			mtos:show_documentation(laptop.modpath.."/apps/os_dialogs_file_selection_help.txt",
					"Help about file selection dialog")
```

Because I like how the text view is solved in mail app in last changes, the development requires #102 so the themes enhancement needs to be finalized first

**Comment by bell07** [comment]
The "Ingame documentation" discussion should be in "Browser" idea issue #67.
The documentation could be the first "Website" in app

Back to the initial idea for this issue

Is someone feels missioned to create the first wiki page in real-life at
 https://github.com/Gerold55/minetest-laptop/wiki ?
